core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -1832,83 +1835,129 @@ public void testShareAutoOffsetResetByDurationInvalidFormat() throws Exception {
+                        curShareCoordNodeId = admin.describeTopics(List.of(Topic.SHARE_GROUP_STATE_TOPIC_NAME)).allTopicNames().get().get(Topic.SHARE_GROUP_STATE_TOPIC_NAME)
+                            .partitions().stream()
+                            .filter(info -> info.partition() == shareGroupStateTp)
+                            .map(info -> info.leader().id())
+                            .toList();
+                    } catch (Exception e) {
+                        fail(e);
+                    }
+                    assertEquals(1, curShareCoordNodeId.size());
+
+                    // shutdown the coordinator
+                    KafkaBroker broker = cluster.brokers().get(curShareCoordNodeId.get(0));
+                    cluster.shutdownBroker(curShareCoordNodeId.get(0));
+
+                    // give some breathing time
+                    broker.awaitShutdown();
+
+                    List<Integer> newShareCoordNodeId = null;
+                    try {
+                        newShareCoordNodeId = admin.describeTopics(List.of(Topic.SHARE_GROUP_STATE_TOPIC_NAME)).allTopicNames().get().get(Topic.SHARE_GROUP_STATE_TOPIC_NAME)
+                            .partitions().stream()
+                            .filter(info -> info.partition() == shareGroupStateTp)
+                            .map(info -> info.leader().id())
+                            .toList();
+                    } catch (Exception e) {
+                        fail(e);
+                    }
 
-            // shutdown the coordinator
-            KafkaBroker broker = cluster.brokers().get(curShareCoordNodeId.get(0));
-            cluster.shutdownBroker(curShareCoordNodeId.get(0));
+                    assertEquals(1, newShareCoordNodeId.size());
+                    assertNotEquals(curShareCoordNodeId.get(0), newShareCoordNodeId.get(0));
+                }, 5L, TimeUnit.SECONDS
+            );
 
-            // give some breathing time
-            broker.awaitShutdown();
+            service.schedule(() -> {
+                    // stop the produce
+                    prodState.done().set(true);
+                }, 10L, TimeUnit.SECONDS
+            );
 
-            List<Integer> newShareCoordNodeId = admin.describeTopics(List.of(Topic.SHARE_GROUP_STATE_TOPIC_NAME)).allTopicNames().get().get(Topic.SHARE_GROUP_STATE_TOPIC_NAME)
-                .partitions().stream()
-                .filter(info -> info.partition() == shareGroupStateTp)
-                .map(info -> info.leader().id())
-                .toList();
+            TestUtils.waitForCondition(
+                () -> prodState.done().get() && consState.done().get(),
+                45_000L,
+                500L,
+                () -> "prod/cons not done yet"
+            );
 
-            assertEquals(1, newShareCoordNodeId.size());
-            assertNotEquals(curShareCoordNodeId.get(0), newShareCoordNodeId.get(0));
+            assertTrue(prodState.count().get() <= consState.count().get());
```

Comment:
And we are making sure that we consumed at least as many records as were produced, not worrying about which records were redelivered (which I think is a perfectly good predicate for this test).